### PR TITLE
kubectl-neat: init at 2.0.3

### LINKED ISSUE #257424: Package request: kubectl-neat
**Project description**

kubectl plugin to remove clutter from Kubernetes manifests to make them more readable.

**Metadata**

* homepage URL:  https://github.com/itaysk/kubectl-neat
* source URL: https://github.com/itaysk/kubectl-neat
* license: Apache License 2.0
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by koralowiec**
## Description of changes

https://github.com/itaysk/kubectl-neat - Kubectl plugin. Clean up Kubernetes yaml and json output to make it readable

This PR should resolve #257424 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by cafkafk — COMMENTED** [review]
about this https://github.com/NixOS/nixpkgs/pull/258395/checks?check_run_id=17292775692 apparently the disk is full, I mentioned it in the ofborg room

**Comment by cole-h** [comment]
@ofborg build kubectl-neat kubectl-neat.passthru.tests

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -33395,6 +33395,8 @@ with pkgs;
 
   kubectl-ktop = callPackage ../applications/networking/cluster/kubectl-ktop { };
 
+  kubectl-neat = callPackage ../applications/networking/cluster/kubectl-neat { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:33398]: Please move the package to `pkgs/by-name/ku/kubectl-neat/package.nix` and simply drop this line.

File: pkgs/applications/networking/cluster/kubectl-neat/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildGoModule, fetchFromGitHub, pkgs }:
```
Comment (wegank) [inline pkgs/applications/networking/cluster/kubectl-neat/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, bash }:
```

File: pkgs/applications/networking/cluster/kubectl-neat/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  vendorHash = "sha256-vGXoYR0DT9V1BD/FN/4szOal0clsLlqReTFkAd2beMw=";
+
+  nativeBuildInputs = [
+    pkgs.bash
+  ];
+
+  postBuild = ''
+    # Replace path to bash in a script
+    # Without this change, there's a problem when running tests
+    sed 's,#!/bin/bash,#!${pkgs.bash}/bin/bash,' -i test/kubectl-stub
```
Comment (wegank) [inline pkgs/applications/networking/cluster/kubectl-neat/default.nix:23]:
```suggestion
    sed 's,#!/bin/bash,#!${bash}/bin/bash,' -i test/kubectl-stub
```

File: pkgs/applications/networking/cluster/kubectl-neat/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    repo = "kubectl-neat";
+    rev = "v${version}";
+    hash = "sha256-j8v0zJDBqHzmLamIZPW9UvMe9bv/m3JUQKY+wsgMTFk=";
+  };
+
+  vendorHash = "sha256-vGXoYR0DT9V1BD/FN/4szOal0clsLlqReTFkAd2beMw=";
+
+  nativeBuildInputs = [
+    pkgs.bash
+  ];
```
Comment (wegank) [inline pkgs/applications/networking/cluster/kubectl-neat/default.nix:18]: Why is this necessary?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (koralowiec) [reply pkgs/top-level/all-packages.nix]: :heavy_check_mark: Moved the package. 

Could provide me more context on this change though? New packages shouldn't be added to [pkgs/top-level/all-packages.nix](https://github.com/NixOS/nixpkgs/pull/258395/files#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22) anymore

File: pkgs/applications/networking/cluster/kubectl-neat/default.nix (hunk 4 above)
Comment (koralowiec) [reply pkgs/applications/networking/cluster/kubectl-neat/default.nix]: But I've just pushed the code without it - I checked that it's not required, but to be honest I'm not sure why

Comment (wegank) [reply pkgs/applications/networking/cluster/kubectl-neat/default.nix]: I'll expect `bash` to be part of `stdenv`, and `buildGo121Module` to be based on `stdenv.mkDerivation`.

Comment (koralowiec) [reply pkgs/applications/networking/cluster/kubectl-neat/default.nix]: Ah, okay. I see that there's a lot of stuff I need to learn about Nix :smile: 
Thanks for the review and additional information!
